File: src/main/java/aurocosh/divinefavor/common/lib/TickCounterTest.java
package aurocosh.divinefavor.common.lib;

import aurocosh.divinefavor.common.util.UtilMath;

public class TickCounterTest {
    private int tickRate;
    private int currentTicks;

    public TickCounterTest() {
        tickRate = 1;
        currentTicks = 0;
    }

    public TickCounterTest(int tickRate) {
        this.tickRate = tickRate;
        currentTicks = 0;
    }

    public void setTickRate(int tickRate) {
        this.tickRate = Math.max(1, Math.abs(tickRate));
    }

    public void setCurrentTicks(int ticks) {
        currentTicks = UtilMath.clamp(ticks, 0, tickRate);
    }

    public boolean tick() {
        if (currentTicks < tickRate) {
            currentTicks++;
            return false;
        }
        else {
            currentTicks = 0;
            return true;
        }
    }
}

File: src/main/java/aurocosh/divinefavor/common/lib/TickCounterTest.kt
package aurocosh.divinefavor.common.lib

import aurocosh.divinefavor.common.util.UtilMath

class TickCounterTest {
    private var tickRate: Int = 0
    private var currentTicks: Int = 0

    constructor() {
        tickRate = 1
        currentTicks = 0
    }

    constructor(tickRate: Int) {
        this.tickRate = tickRate
        currentTicks = 0
    }

    fun setTickRate(tickRate: Int) {
        this.tickRate = Math.max(1, Math.abs(tickRate))
    }

    fun setCurrentTicks(ticks: Int) {
        currentTicks = UtilMath.clamp(ticks, 0, tickRate)
    }

    fun tick(): Boolean {
        if (currentTicks < tickRate) {
            currentTicks++
            return false
        } else {
            currentTicks = 0
            return true
        }
    }
}
